frontend/ui/screensaver.lua

```diff
@@ -788,4 +793,19 @@ function Screensaver:close()
     end
 end
 
+function Screensaver:dtor()
```

Comment:
wtf? lol! yoro! _You only read once_, so please make this readable :)
Also, we don't have any occurence of `:dtor()` or `:destructor()`, we usually do that kind of stuff in a `:free()`

Also `FileConverter:dtor()`

Reply:
I wasn't too keen on `free` as we don't actually free anything, we just just flag it for collection by the GC ;).

Reply:
Still not fond of `:destructor()`. Our method names are usually verbs. `:destroy()` would be more in line. But if you don't actually free anything, you don't actually destroy anything :)
I'm sure we have other `:free()` that don't do any free - it's just the classic method name that you call when you're done with an object and want stuff freed now - which is what you want, even if you don't do any real C free().

Reply:
I don't want to actually stomp on a real `free` method, though, and I wanted this to be "portable" ;).

Reply:
I could use `fini`, which pairs nicely with the semantics of `init`, but you're probably not going to like it either ;o).

Reply:
So `:release()`, `:cleanup()` and stick on that name for future methods to do some _cleanup without stomping on a real free_ ?
(But this fact and new names will make it harder to know what you should call/name your stuff - _if I don't name it `:free()`, it won't be called by UIManager/onclose/oncloseWidget() - so should I name my stuff :free() or :release() ?_ Is it really worth adding this and making this distinction ? May be this one should call any upper free() to avoid stomping ?

I also quickly read over your getmetatable/super stuff - it looked odd and new, but well, didn't care. May be if you need this kind of stuff, it's that the whole thing is crap, and need being fully rewritten ? :)

Reply:
> the whole thing is crap, and need being fully rewritten

That is what I ended up doing w/ `filemanagersetdefaults` ;p.

Snark aside, it *might* come in handy in some very specific circumstances, but, yeah, in those two instances it's arguably a case of broken by design ;).

Reply:
> I could use `fini`, which pairs nicely with the semantics of `init`, but you're probably not going to like it either ;o).

Indeed no :) We have a good corpus of code, which is good litterature. No need to make it look like foreign code :)

Reply:
(Went with `cleanup`, FWIW).